### PR TITLE
feat: add C implementation for `math/base/special/roundb`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/roundb`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/math/base/special/roundb).

```c
double stdlib_base_roundb( const double x, const double n, const double b );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@gunjjoshi Would it make more sense to make `n` and `b` 32-bit integers, similar to `n` in `roundn`?

**Comment by gunjjoshi** [comment]
> @gunjjoshi Would it make more sense to make `n` and `b` 32-bit integers, similar to `n` in `roundn`?

Sure @kgryte, I will change that.

**Comment by gunjjoshi** [comment]
@kgryte, Do we have `STDLIB_MATH_BASE_NAPI_MODULE_DII_D( stdlib_base_roundb )`? Or should we use something else here?

Edit: checked it, we don't have this.

**Comment by gunjjoshi** [comment]
Should I add `napi_value stdlib_math_base_napi_dii_d( napi_env env, napi_callback_info info, double (*fcn)( double, int32_t, int32_t ) )` in [`math/base/napi/ternary/src/main.c`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/napi/ternary/src/main.c) in a seperate PR?
And then we can use it here.

**Comment by kgryte** [comment]
@gunjjoshi Yes, that would be great!

**Comment by kgryte** [comment]
I believe you'll also need to add a corresponding macro to that same package.

**Comment by gunjjoshi** [comment]
Yes. I'll add the macro in the `.h` file as well.

**Comment by gunjjoshi** [comment]
@kgryte Our [`math/base/assert/is-nan` ](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-nan) and  [`math/base/assert/is-infinite` ](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-infinite) take `double` as input. But we have 2 integer values, `n` and `b` to handle here. We have some test cases which fail due to this. I checked [`roundn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/roundn), but over there, we do not have test cases which give `NaN` or `+-infinite` as the arguments for the integer `n`.

**Comment by kgryte** [comment]
As discussed offline, the plan here is to remove non-applicable tests for the native bindings for `n` and `b` with regard to the C implementation and its restriction to only accept integer arguments.

**Comment by gunjjoshi** [comment]
I have also removed the checks in `main.c` where we check if `b` or `n` is `NaN` or `infinite`.

File: lib/node_modules/@stdlib/math/base/special/roundb/examples/c/example.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    const double x[] = { -5.0, -3.89, -2.78, -1.67, -0.56, 0.56, 1.67, 2.78, 3.89, 5.0 };
+	const int32_t n[] = { -4, -3, -2, -1, 0, 1, 2, 3, 4, 5 };
+	const int32_t b[] = { 20, 19, 18, 17, 16, 15, 14, 13, 12, 11 };
+
+    double v;
+    int i;
+    for ( i = 0; i < 10; i++ ) {
+        v = stdlib_base_roundb( x[ i ], n[ i ], b[ i ] );
+        printf( "roundb(%lf, %d, %d) = %lf\n", x[ i ], n[ i ], b[ i ], v );
+    }
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/roundb/examples/c/example.c]: Should use tab-indentation: 

```suggestion
	const double x[] = { -5.0, -3.89, -2.78, -1.67, -0.56, 0.56, 1.67, 2.78, 3.89, 5.0 };
	const int32_t n[] = { -4, -3, -2, -1, 0, 1, 2, 3, 4, 5 };
	const int32_t b[] = { 20, 19, 18, 17, 16, 15, 14, 13, 12, 11 };

	double v;
	int i;
	for ( i = 0; i < 10; i++ ) {
		v = stdlib_base_roundb( x[ i ], n[ i ], b[ i ] );
		printf( "roundb(%lf, %d, %d) = %lf\n", x[ i ], n[ i ], b[ i ], v );
	}
```

**Review by Planeshifter — APPROVED** [review]
Thanks, @gunjjoshi! Will merge once CI passes.

**Pull request merged**